flink-table/flink-table-common/src/test/java/org/apache/flink/table/types/inference/TypeStrategiesTest.java

```diff
@@ -311,7 +315,21 @@
                                 "Average without grouped aggregation",
                                 TypeStrategies.aggArg0(LogicalTypeMerging::findAvgAggType, true))
                         .inputTypes(DataTypes.INT().notNull())
-                        .expectDataType(DataTypes.INT()));
+                        .expectDataType(DataTypes.INT()),
+
+                // CURRENT_WATERMARK
+                TestSpec.forStrategy("TIMESTAMP(3) *ROWTIME*", TypeStrategies.CURRENT_WATERMARK)
```

Comment:
I couldn't really find a grouped cut that made sense (other than the decimal ones), so I moved all specific strategies into their own class now and split up the test as well. I settled for `INSTANCE` members to not add them all to `TypeStrategies`, but we can also undo that. Let me know what you think of this, I added it as a separate commit (also since it's kind of unrelated and more drive-by refactoring).